### PR TITLE
fix: mark ModelConvertor parameters and returns as non-null (#2507)

### LINKED ISSUE #2507: ModelConvertor interface should guarantee its method parameters cannot be null
```
SolverModel_ toSolverModel(ModelInput_ modelInput, ModelConfig<ModelConfigurationOverrides_> modelConfig, Optional<ModelOutput_> lastModelOutput);
```

(and other methods as well) does not provide clear information about nullity of its parameters.

### PR DESCRIPTION
**Pull request by vikas-prasad-cx**
## Summary
- Mark `ModelConvertor` with `@NullMarked` and document that method parameters/returns are never null (use empty `Optional` for no prior output).
- Align `AbstractTrivialModelConvertor` with the same contract via runtime `Objects.requireNonNull` checks, removing the previous null-tolerant `modelOutput` path.
- Add unit tests covering the annotation contract and null-argument rejection.

Fixes #2507

## Test plan
- [x] `./mvnw test -pl service/definition -Dtest=ModelConvertorNullabilityTest`
- [x] `./mvnw test -pl service/quarkus/deployment -Dtest=AbstractTrivialModelConvertorTest`
- [x] `./mvnw test -pl service/definition,service/quarkus/deployment` (40 tests passed)
- [ ] CI checks on this PR

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by vikas-prasad-cx** [comment]
Hi — gentle ping on this PR when you have a chance. Happy to address any feedback. CI may still need first-time contributor workflow approval. Thanks!

**Comment by triceo** [comment]
Hello @vikas-prasad-cx - we are aware of your PR and we will get to it when time permits.
Thank you for your contribution, and for your patience.

File: service/definition/src/test/java/ai/timefold/solver/service/definition/api/ModelConvertorNullabilityTest.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package ai.timefold.solver.service.definition.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.jspecify.annotations.NullMarked;
+import org.junit.jupiter.api.Test;
+
+class ModelConvertorNullabilityTest {
```
Comment (rsynek) [inline service/definition/src/test/java/ai/timefold/solver/service/definition/api/ModelConvertorNullabilityTest.java]: Not sure this test pulls its weight; once the annotation is there, nobody is going to remove it without a good reason.

**Review by rsynek — CHANGES_REQUESTED** [review]
@vikas-prasad-cx Thanks for your contribution!

Also, my apologies for the delay with reviewing your PR.

I left one comment inline, plus, in order to declare non-null parameters of the `ModelConvertor` methods we must ensure that the caller never passes nulls there.

In case we don't have the `ModelConfig` instance, there is [empty factory method](https://github.com/TimefoldAI/timefold-solver/blob/main/service/definition/src/main/java/ai/timefold/solver/service/definition/api/domain/ModelConfig.java#L16).

**Comment by vikas-prasad-cx** [comment]
@rsynek Thank you for the review, and no worries about the delay.

Addressed in 8895be2:

- Removed `ModelConvertorNullabilityTest` as requested.
- `SolverWorker` now passes `ModelConfig.empty()` when no config is available, so `ModelConvertor` callers never hand in a null `ModelConfig`. The existing `AbstractTrivialModelConvertor` null-checks and the `ModelConfig.empty()` happy-path test are unchanged.

Happy to adjust further if needed.

File: service/definition/src/test/java/ai/timefold/solver/service/definition/api/ModelConvertorNullabilityTest.java (hunk 1 above)
Comment (vikas-prasad-cx) [reply service/definition/src/test/java/ai/timefold/solver/service/definition/api/ModelConvertorNullabilityTest.java]: Removed this test as requested.

**Review by rsynek — APPROVED** [review]

**Comment by rsynek** [comment]
> @rsynek Thank you for the review, and no worries about the delay.
> 
> Addressed in [8895be2](https://github.com/TimefoldAI/timefold-solver/commit/8895be2df5a2649d6f3b02e7b2e802f23093201b):
> 
> * Removed `ModelConvertorNullabilityTest` as requested.
> * `SolverWorker` now passes `ModelConfig.empty()` when no config is available, so `ModelConvertor` callers never hand in a null `ModelConfig`. The existing `AbstractTrivialModelConvertor` null-checks and the `ModelConfig.empty()` happy-path test are unchanged.
> 
> Happy to adjust further if needed.

Thank you, looks very good now! I will take care of the CI checks and let you if any further change is needed.
